apps/sms/views/shared/js/shim_host_bootstrap.js

```diff
@@ -7,6 +8,12 @@
 'use strict';
 
 exports.bootstrap = function(appInstanceId) {
+  // We can't handle system messages inside iframe right now, so we access
+  // parent window here. This should be removed once bug 818000 is resolved.
+  ActivityShim.init(
```

Comment:
Yeah, as expected, there is no any noticeable improvement per raptor, do you have any concerns regarding moving ActivityShim to iframe?